ua-snap/epa-justice
# Aggregation

Apologies in advance for all the `black` reformatting changes! You can safely ignore a lot of the diff.

The real heart of this PR (and the only major change from the previously reviewed work) is the `functions.aggregate_results()` function. This function takes the places with multiple geographic units (currently only JBER and Eagle River) and aggregates the values according to the guidance from the PI. The operation requires converting the values from percentages back to population counts, then summing the counts and re-computing the percentage. GEOID strings and place names of aggregated rows are also concatenated.

I also incorporated guidance from the PI about dealing with NA values. The resulting `data_to_export.csv` now has way more actual data values and fewer NAs. Some new race / ethnicity variables were added as well.

Same as the last PR, I called a few of the individual data fetching functions in `fetch_data_and_export.ipynb` and included some URL request printing to allow you to see the API requests, the JSON returned, and compare them with the reformatted function outputs.

TO TEST:

- Read the updated `README.md`, check for clarity / typos / etc.
- Run `fetch_data_and_export.ipynb`, paying particular attention to the testing cells in the first part of the notebook. Investigate some of the URLs, and make sure the values you see in the printed dataframes match those returned by the API. (I recently ran into some timeout issues with the API, but it seemed to resolve itself and was working normally within a few hours. If you time out, or recieve any other errors when running the `run_fetch_and_merge()` function in the notebook, maybe wait an hour and try again.)
- Review the code for the `functions.aggregate_results()` function, confirm that the aggregation math makes sense.
 - Review `data_to_export.csv`. The result should be identical to the version committed in this branch.
